canonical/knative-operators
# add juju config to use custom images for knative eventing

This adds to knative-eventing:

* the `custom_images` config to to the charm, allowing users to provide overrides to the default images so they can use their own (for example, if they work in an air-gapped environment)
* `DEFAULT_IMAGES` defined in the charm that allow the charm to have an opinion on what images should be used if none are specified
* tests for the above features

### Manual test instructions
```
cd charms/knative-eventing
charmcraft pack
juju deploy knative-operator --trust --channel latest/edge
juju deploy ./knative-eventing_ubuntu-20.04-amd64.charm --trust --config namespace=knative-eventing
kubectl get pod -n knative-eventing  # observe the knative eventing pods
juju config knative-eventing custom_images="{eventing-controller: not-a-real-image}"
kubectl get pod -n knative-eventing  # observe how a new eventing-controller pod is started.  This will not deploy successfully as the image above is not real
juju config knative-eventing custom_images=""  # reset back to defaults
kubectl get pod -n knative-eventing  # observe the knative eventing-controller pods
```
